hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -412,6 +412,43 @@ private void validateTableIsIcebergTableOrView(
     }
   }
 
+  /**
+   * Check whether table or metadata table exists.
+   *
+   * <p>Note: If a hive table with the same identifier exists in catalog, this method will return
+   * {@code false}.
+   *
+   * @param identifier a table identifier
+   * @return true if the table exists, false otherwise
+   */
+  @Override
+  public boolean tableExists(TableIdentifier identifier) {
+    TableIdentifier baseTableIdentifier = identifier;
+    if (!isValidIdentifier(identifier)) {
+      if (!isValidMetadataIdentifier(identifier)) {
+        return false;
```

Comment:
Minor behavior issue here.  Currently if it's an invalid identifier, it will throw as opposed to returning false.

```      
throw new NoSuchTableException("Invalid table identifier: %s", identifier);
```

Reply:
thank you @danielcweeks , I think you might refer to [this line in loadTable()](https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java#L66) which throw exceptions, for the [`tableExists` API defined in catalog interface](https://github.com/apache/iceberg/blob/main/api/src/main/java/org/apache/iceberg/catalog/Catalog.java#L283-L285) actually catch this NoSuchTableException and return false. 

I also added a explicit step in unit test for invalid identifier to ensure the behavior is same as before